### ISSUE TITLE
pipeline.sh: missing files?

### ISSUE THREAD

**Issue by ckindler123** [MASILab/Synb0-DISCO#55]
Hi there,

We tried to get Synb0-DISCO run using docker and got an error message (see at the bottom of the email). The script pipeline.sh uses the files /extra/dual_channel_unet/num_fold_"$i"total_folds"$NUM_FOLDS"seed_1_num_epochs_100_lr_0.0001_betas(0.9,\ 0.999)weight_decay_1e-05_num_epoch*.pth
We do not know, what is the path to the files in docker, but in the folders in Github we found the files in the folder src/train_lin. Could that be the reason for pipeline.sh not to find the files and how to fix it?

Thank you for your help!

Copying results to results path...
-------
Removing job directory...
Performing inference on FOLD: 1
/extra/pipeline.sh: line 42:  1194 Killed                  python3.6 /extra/inference.py /OUTPUTS/T1_norm_lin_atlas_2_5.nii.gz /OUTPUTS/b0_d_lin_atlas_2_5.nii.gz /OUTPUTS/b0_u_lin_atlas_2_5_FOLD_"$i".nii.gz /extra/dual_channel_unet/num_fold_"$i"_total_folds_"$NUM_FOLDS"_seed_1_num_epochs_100_lr_0.0001_betas_\(0.9\,\ 0.999\)_weight_decay_1e-05_num_epoch_*.pth
Performing inference on FOLD: 2
/extra/pipeline.sh: line 42:  1394 Killed                  python3.6 /extra/inference.py /OUTPUTS/T1_norm_lin_atlas_2_5.nii.gz /OUTPUTS/b0_d_lin_atlas_2_5.nii.gz /OUTPUTS/b0_u_lin_atlas_2_5_FOLD_"$i".nii.gz /extra/dual_channel_unet/num_fold_"$i"_total_folds_"$NUM_FOLDS"_seed_1_num_epochs_100_lr_0.0001_betas_\(0.9\,\ 0.999\)_weight_decay_1e-05_num_epoch_*.pth
Performing inference on FOLD: 3
/extra/pipeline.sh: line 42:  1594 Killed                  python3.6 /extra/inference.py /OUTPUTS/T1_norm_lin_atlas_2_5.nii.gz /OUTPUTS/b0_d_lin_atlas_2_5.nii.gz /OUTPUTS/b0_u_lin_atlas_2_5_FOLD_"$i".nii.gz /extra/dual_channel_unet/num_fold_"$i"_total_folds_"$NUM_FOLDS"_seed_1_num_epochs_100_lr_0.0001_betas_\(0.9\,\ 0.999\)_weight_decay_1e-05_num_epoch_*.pth
Performing inference on FOLD: 4
Performing inference on FOLD: 5
/extra/pipeline.sh: line 42:  1794 Killed                  python3.6 /extra/inference.py /OUTPUTS/T1_norm_lin_atlas_2_5.nii.gz /OUTPUTS/b0_d_lin_atlas_2_5.nii.gz /OUTPUTS/b0_u_lin_atlas_2_5_FOLD_"$i".nii.gz /extra/dual_channel_unet/num_fold_"$i"_total_folds_"$NUM_FOLDS"_seed_1_num_epochs_100_lr_0.0001_betas_\(0.9\,\ 0.999\)_weight_decay_1e-05_num_epoch_*.pth
/extra/pipeline.sh: line 42:  1994 Killed                  python3.6 /extra/inference.py /OUTPUTS/T1_norm_lin_atlas_2_5.nii.gz /OUTPUTS/b0_d_lin_atlas_2_5.nii.gz /OUTPUTS/b0_u_lin_atlas_2_5_FOLD_"$i".nii.gz /extra/dual_channel_unet/num_fold_"$i"_total_folds_"$NUM_FOLDS"_seed_1_num_epochs_100_lr_0.0001_betas_\(0.9\,\ 0.999\)_weight_decay_1e-05_num_epoch_*.pth
Taking ensemble average
Image Exception : #63 :: No image files match: /OUTPUTS/b0_u_lin_atlas_2_5_FOLD_*
terminate called after throwing an instance of 'std::runtime_error'
  what():  No image files match: /OUTPUTS/b0_u_lin_atlas_2_5_FOLD_*
/extra/pipeline.sh: line 49:  2194 Aborted                 fslmerge -t /OUTPUTS/b0_u_lin_atlas_2_5_merged.nii.gz /OUTPUTS/b0_u_lin_atlas_2_5_FOLD_*.nii.gz
Image Exception : #63 :: No image files match: /OUTPUTS/b0_u_lin_atlas_2_5_merged
terminate called after throwing an instance of 'std::runtime_error'
  what():  No image files match: /OUTPUTS/b0_u_lin_atlas_2_5_merged
/extra/pipeline.sh: line 50:  2195 Aborted                 fslmaths /OUTPUTS/b0_u_lin_atlas_2_5_merged.nii.gz -Tmean /OUTPUTS/b0_u_lin_atlas_2_5.nii.gz
Applying inverse xform to undistorted b0
 file /OUTPUTS/b0_u_lin_atlas_2_5.nii.gz does not exist . 
terminate called after throwing an instance of 'itk::ExceptionObject'
  what():  /home/local/VANDERBILT/blaberj/ANTS_13_FEB_2019/bin/ants/ITKv5/Modules/Core/Common/src/itkProcessObject.cxx:1412:
itk::ERROR: ResampleImageFilter(0x2e5c700): Input Primary is required but not set.
/extra/pipeline.sh: line 54:  2196 Aborted                 antsApplyTransforms -d 3 -i /OUTPUTS/b0_u_lin_atlas_2_5.nii.gz -r /INPUTS/b0.nii.gz -n BSpline -t [/OUTPUTS/epi_reg_d_ANTS.txt,1] -t [/OUTPUTS/ANTS0GenericAffine.mat,1] -o /OUTPUTS/b0_u.nii.gz
Applying slight smoothing to distorted b0
Running topup
Image Exception : #63 :: No image files match: /OUTPUTS/b0_u
terminate called after throwing an instance of 'std::runtime_error'
  what():  No image files match: /OUTPUTS/b0_u
/extra/pipeline.sh: line 65:  2207 Aborted                 fslmerge -t /OUTPUTS/b0_all.nii.gz /OUTPUTS/b0_d_smooth.nii.gz /OUTPUTS/b0_u.nii.gz
Image Exception : #63 :: No image files match: /OUTPUTS/b0_all
Image Exception : #22 :: Failed to read volume /OUTPUTS/b0_all.nii.gz
Error : No image files match: /OUTPUTS/b0_all

Part of FSL (ID: 6.0.1)
topup

Usage: 
topup --imain=<some 4D image> --datain=<text file> --config=<text file with parameters> --out=my_topup_results

Compulsory arguments (You MUST set one or more of):
	--imain		name of 4D file with images
	--datain	name of text file with PE directions/times

Optional arguments (You may optionally specify one or more of):
	--out		base-name of output files (spline coefficients (Hz) and movement parameters)
	--fout		name of image file with field (Hz)
	--iout		name of 4D image file with unwarped images
	--logout	Name of log-file
	--warpres	(approximate) resolution (in mm) of warp basis for the different sub-sampling levels, default 10
	--subsamp	sub-sampling scheme, default 1
	--fwhm		FWHM (in mm) of gaussian smoothing kernel, default 8
	--config	Name of config file specifying command line arguments
	--miter		Max # of non-linear iterations, default 5
	--lambda	Weight of regularisation, default depending on --ssqlambda and --regmod switches. See user documetation.
	--ssqlambda	If set (=1), lambda is weighted by current ssq, default 1
	--regmod	Model for regularisation of warp-field [membrane_energy bending_energy], default bending_energy
	--estmov	Estimate movements if set, default 1 (true)
	--minmet	Minimisation method 0=Levenberg-Marquardt, 1=Scaled Conjugate Gradient, default 0 (LM)
	--splineorder	Order of spline, 2->Qadratic spline, 3->Cubic spline. Default=3
	--numprec	Precision for representing Hessian, double or float. Default double
	--interp	Image interpolation model, linear or spline. Default spline
	--scale		If set (=1), the images are individually scaled to a common mean, default 0 (false)
	--regrid		If set (=1), the calculations are done in a different grid, default 1 (true)
	-h,--help	display help info
	-v,--verbose	Print diagonostic information while running
	-h,--help	display help info

Failed to read volume /OUTPUTS/b0_all.nii.gz
Error : No image files match: /OUTPUTS/b0_all
FINISHED!!!

**Comment by MattRicchi** [comment]
Hello, 
I'm having the same issue. Did you manage to solve it by any chance? 

Thank youfor your help!

**Comment by ckindler123** [comment]
Hi Mattia,

 

yes, we could solve the problem! It was the RAM - the command needs more than 16 RAM and abort when the 16 RAM are reached. Now it works on a another laptop with 32 RAM. Hope it helps!

 

Best wishes

Christine

 
 

Gesendet: Donnerstag, 09. November 2023 um 16:01 Uhr
Von: "Mattia Ricchi" ***@***.***>
An: "MASILab/Synb0-DISCO" ***@***.***>
Cc: "ckindler123" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [MASILab/Synb0-DISCO] pipeline.sh: missing files? (Issue #55)


 

Hello,
I'm having the same issue. Did you manage to solve it by any chance?

Thank youfor your help!

**Comment by BennettLandman** [comment]
I am closing the older bug reports as these were missed. We are now better tracking reports across the organization. Please re-open if this continues to be a blocker.
